microsoft/testsuites/core/timesync.py

```diff
@@ -64,3 +68,45 @@ def timesync_validate_ptp(self, environment: Environment, node: Node) -> None:
                 assert_that(chrony_results.stdout).described_as(
                     "Chrony config file should use the symlink /dev/ptp_hyperv."
                 ).contains(self.hyperv_ptp_udev_rule)
+
+    @TestCaseMetadata(
+        description="""
+        This test is to check -
+            1. Check clock source name shown as hyperv_clocksource_tsc_page, except:
+                1.1 If it is centos/rhel/oracle below 7.5 + with LIS driver, it
+                    shown as lis_hv_clocksource_tsc_page.
+                1.2 If it is centos/rhel/oracle below 7.2 + without LIS driver, it
+                    shown as hyperv_clocksource.
+            2. Check CPU flag contains tsc from /proc/cpuinfo.
+            3. Check clocksource name shown up in dmesg.
+        """,
+        priority=2,
+    )
+    def timesync_check_timesource(self, environment: Environment, node: Node) -> None:
+        # 1. Check clock source name shown as hyperv_clocksource_tsc_page
+        clocksource = "hyperv_clocksource_tsc_page"
+        cat = node.tools[Cat]
+        clock_source_result = cat.run(self.current_clocksource)
+        os_version = node.os.os_version
+        os_verion_convert = (
+            os_version.release.split(".")[0] + "." + os_version.release.split(".")[1]
+        )
+        if isinstance(node.os, Redhat):
+            if node.os._package_exists("kmod-microsoft"):
```

Comment:
I have a commit which is adding LIS driver as a tool and can get_version() to check if LIS drivers are installed or not. May be we can use that tool if that commit gets merged.

Reply:
sure

Reply:
I found the lisdirver tool is not suitable here, once this tool is invoked, it will try to install it.

Reply:
Hmm we can may be add a line to check_exist() before installing it in the _install() method.
Tis will let it not install if already installed.

Reply:
current logic is like this, when not exist, it will install it, but in my case, I just want to judge it exists or not, no need to install it.